apps/speed.c

```diff
@@ -3950,12 +3950,23 @@ int speed_main(int argc, char **argv)
     }
 
     for (k = 0; k < ALGOR_NUM; k++) {
+        const char *alg_name;
+
         if (!doit[k])
             continue;
+
+        if (k == D_EVP) {
+            if (evp_cipher == NULL)
+                alg_name = evp_md_name;
+            else if ((alg_name = EVP_CIPHER_get0_name(evp_cipher)) == NULL)
+                app_bail_out("failed to get name of cipher '%s'\n", evp_cipher);
+        } else
+            alg_name = names[k];
```

Comment:
Our code style would include {} around the else branch (because the if part has {}).

Reply:
Fixed in new PR version